packages/components/src/navigation/index.js

```diff
@@ -1,74 +1,53 @@
  */
-import { useState } from '@wordpress/element';
-import { Icon, arrowLeft } from '@wordpress/icons';
+import { useEffect, useState } from '@wordpress/element';
 
-/**
- * Internal dependencies
- */
-import Button from '../button';
-import Text from '../text';
-import Item from './item';
+const Navigation = ( { activeId, children, items: rawItems, rootTitle } ) => {
+	const [ activeLevel, setActiveLevel ] = useState( 'root' );
+
+	const mapItemData = ( items ) => {
+		return items.map( ( item ) => {
+			const itemChildren = rawItems.filter(
+				( i ) => i.parent === item.id
+			);
+			return {
+				...item,
+				children: itemChildren,
+				parent: item.parent || 'root',
+				isActive: item.id === activeId,
+				hasChildren: itemChildren.length > 0,
+			};
+		} );
+	};
+	const items = [
+		{ id: 'root', title: rootTitle },
+		...mapItemData( rawItems ),
+	];
 
-const Navigation = ( { data, initial } ) => {
-	const initialActive = data.find( ( item ) => item.slug === initial );
-	const [ active, setActive ] = useState( initialActive );
-	const parent = data.find( ( item ) => item.slug === active.parent );
-	const items = data.filter( ( item ) => item.parent === active.parent );
+	const activeItem = items.find( ( item ) => item.id === activeId );
+	const level = items.find( ( item ) => item.id === activeLevel );
+	const levelItems = items.filter( ( item ) => item.parent === level.id );
+	const parentLevel =
+		level.id === 'root'
+			? null
+			: items.find( ( item ) => item.id === level.parent );
 
-	const goBack = () => {
-		if ( ! parent.parent ) {
-			// We are at top level, will need to handle this case.
-			return;
-		}
-		const parentalSiblings = data.filter(
-			( item ) => item.parent === parent.parent
-		);
-		if ( parentalSiblings.length ) {
-			setActive( parentalSiblings[ 0 ] );
+	useEffect( () => {
+		if ( activeItem ) {
+			setActiveLevel( activeItem.parent );
 		}
-	};
+	}, [] );
 
 	return (
 		<div className="components-navigation">
-			<Button
-				isSecondary
-				className="components-navigation__back"
-				onClick={ goBack }
-			>
-				<Icon icon={ arrowLeft } />
-				{ parent.back }
-			</Button>
-			<div className="components-navigation__title">
-				<Text variant="title.medium">{ parent.title }</Text>
-			</div>
-			<div className="components-navigation__menu-items">
-				{ items.map( ( item ) =>
-					item.menu !== 'secondary' ? (
-						<Item
-							key={ item.slug }
-							data={ data }
-							item={ item }
-							setActive={ setActive }
-							isActive={ item.slug === active.slug }
-						/>
-					) : null
-				) }
-			</div>
-			<div className="components-navigation__menu-items is-secondary">
-				{ items.map( ( item ) =>
-					item.menu === 'secondary' ? (
-						<Item
-							key={ item.slug }
-							data={ data }
-							item={ item }
-							setActive={ setActive }
-							isActive={ item.slug === active.slug }
-						/>
-					) : null
-				) }
-			</div>
+			{ children( {
+				activeId,
```

Comment:
Lets omit `activeId`. Its the same as the supplied `activeId` prop anyways.

Reply:
Good call.  Removed.